0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,279 @@
+
+Timeouts on the HTLCs are configured globally for the payee node and can optionally be defined when paying by the payer node (`--final_cltv_delta=T`).
+
+
+IMPORTANT: TODO: we should investigate this topic more in detail.
+
+=== Responsibilities
+
+A main goal of COMIT is to keep the autonomy to the user and let him/her decide when to deploy a HTLC, redeem or refund a HTLC, etc.
+If a trade involves LN using LND we can approach these things differently:
+
+
+* Action
+    ** Create hold invoice
+* Responsibility
+    ** LND
+* Invoker
+    ** User, comit-i, comit-node
+* Description
+    ** `addholdinvoice` is available as a RPC command or through the LND CLI. Although dealing with this is rather cumbersome, to keep the autonomy with the user, and to not introduce LND dependency into the comit-node, we this should be possible through comit-i.
+* Conclusion:
+    ** comit-i needs LND support. However, in order to do this, we will need to introduce a new action which is meant to be executed prior accepting a swap request:
+    *** Bob receives a swap request from Alice (an learns about the hashed secret)
+    *** *Action 1:* Bob creates a hold invoice through comit-i
+    *** *Action 2:* Bob accepts the swap requests by posting the newly generated invoice ID back to comit-rs
+    A quick research showed that LN payments can be done with the browser extension: https://lightningjoule.com/[Joule] and requests to a LND node can be done through the browser.
+
+---
+
+
+* Action
+    ** Pay invoice
+* Responsibility:
+    ** LND or LN Wallet
+* Invoker
+    ** User, comit-i
+* Description
+    ** To keep the autonomy to the user when to initiate a trade, we should return the invoice information through our API to the user (e.g. expose it through comit-i ) and let him/her pay the invoice.
+* Conclusion
+    ** comit-i needs LND support.
+
+
+
+---
+
+
+* Action:
+    ** Settle Invoice
+* Responsibility
+    ** LND or LN Wallet
+* Invoker
+    ** User, comit-i, comit-node
+* Description
+    ** As soon as the secret has been learned, the HTLC on the LN should be settled using the command `settleinvoice`, this can either be done by the user (and exposed through comit-i) or done automatically through the comit-node. Since we have the extra _redeem_ step for Bitcoin and Ethereum (as well for Erc20) which needs to be performed from the user, we should leave the settlement of the invoice to the user (e.g. expose this information through comit-i).
+* Conclusion
+    ** comit-i needs LND support.
+
+
+
+---
+
+
+* Action
+    ** Monitor LN
+* Responsibility
+    ** LND
+* Invoker
+    ** Btsieve
+* Description
+    ** Similar to other Ledgers we need to monitor LN for the payment (and later on settlement) of an invoice. To keep our current abstraction layer, this should be done through btsieve
```

Comment:
I disagree. For me `btsieve` is about monitoring L1. L2 doesn't have blocks so none of the abstractions built into `btsieve` fit for L2.

Reply:
I disagree with you. Btsieve knows about blocks and transactions, transactions can either be done on L2 or L1.

Reply:
Like Thomas, I think the idea was that `btsieve` could be run a service. This implies that it only works on public ledgers.